### ISSUE TITLE
I can't get the ngModel to fill in the input

### ISSUE THREAD

**Issue by rr-phillips** [azzamasghar1/ion-intl-tel-input#7]
I have ngModel set, with data in the value, and the js file recognizes it, but then overwrites it with null? i tried to have the placeholder disabled, but that did help. I'm wondering if I'm missing a setting or something?

**Comment by rr-phillips** [comment]
so I got it to fill in, once I set the existing data to be in the format of the object returned by the component {dialCode:...}. None of my data is currently set up that way, and would cause significant issues with my app to change the format at this point. is there a way to get just a string to be used as an ngModel and actually be filled in, or do I have to convert all my data before it will be used to fill in the input. Thank you!

**Comment by JimmehG** [comment]
Would love this answered too. Doesn't seem to work for reactive forms either. Console returns this:
`
vendor.js:347800 ["123"]0: "123"length: 1__proto__: Array(0)
vendor.js:347801 null
vendor.js:347820 do
vendor.js:347725 set value
vendor.js:347726 value: ["123"]0: "123"length: 1__proto__: Array(0)
`
And after this the value in my FormGroup is null.

**Comment by thealiano** [comment]
> As a temporary workaround, I'm using the following to set value:
> 
> `import {PhoneNumber, PhoneNumberFormat, PhoneNumberUtil} from 'google-libphonenumber';`
> 
> ```
> setPhoneNumber() {
>         const DEFAULT_COUNTRY_CODE = 'us';
>         this.visitorForm.get('mobileNumber').setValue(
>             this.parseTelephoneNumberToIonIntlTelValue(
>                 '+12024561414',
>                 DEFAULT_COUNTRY_CODE)
>         );
>     }
> ```
> 
> ```
>     parseTelephoneNumberToIonIntlTelValue(rawNumber, defaultCountryCode) {
>         const probePhoneNumber = this.phoneBookImport.mobileNumber;
>         let googleNumber: PhoneNumber;
>         try {
>             googleNumber = this.phoneUtil.parse(
>                 probePhoneNumber,
>                 defaultCountryCode.toUpperCase()
>             );
>         } catch (e) {
>         }
> 
>         const internationalNo = googleNumber
>             ? this.phoneUtil.format(googleNumber, PhoneNumberFormat.INTERNATIONAL)
>             : '';
>         const nationalNo = googleNumber
>             ? this.phoneUtil.format(googleNumber, PhoneNumberFormat.NATIONAL)
>             : '';
>         const regionCode = googleNumber
>             ? this.phoneUtil.getRegionCodeForNumber(googleNumber).toLowerCase()
>             : defaultCountryCode;
> 
>         return {
>             internationalNumber: internationalNo,
>             nationalNumber: nationalNo,
>             isoCode: regionCode
>         };
>     }
> ```

Hi ,
 i cannot set default value , i want to set the default country code only , i tried your solution and it's works but how can i adapt it to set only the country code and not the number ?

**Comment by metadevj** [comment]
Hi, to set only the default country I'm using:

`
clear() {
        // Work Around >>> Manually Reset ion-intl-tel-input
        this.form.get('mobileNumber').setValue({
            internationalNumber: '',
            nationalNumber: '',
            isoCode: this.defaultCountryCode.toLowerCase()
        });
        // Reset Form
        this.form.reset();
    }
`

**Comment by thealiano** [comment]
> Hi, to set only the default country I'm using:
> 
> ```
> clear() {
>         // Work Around >>> Manually Reset ion-intl-tel-input
>         this.form.get('mobileNumber').setValue({
>             internationalNumber: '',
>             nationalNumber: '',
>             isoCode: this.defaultCountryCode.toLowerCase()
>         });
>         // Reset Form
>         this.form.reset();
>     }
> ```

Nice solution i will try it now , why did you reset the form ? To avoid the trigger of the validators ?

**Comment by metadevj** [comment]
> Nice solution i will try it now , why did you reset the form ? To avoid the trigger of the validators ?

Sorry. this.form.reset() is not required.

**Comment by thealiano** [comment]
Thanks to you. 

Here my final code , working as a charm  👍 
```
this.authForm = this.formBuilder.group({
          phoneNumber: new FormControl(
            {
                internationalNumber: '',
                nationalNumber: '',
                isoCode: this.countryDefault
            }
            , [
            Validators.required,
            IonIntlTelInputValidators.phone
          ])
      );
```
Thank you

**Comment by azzamasghar1** [comment]
There was an issue with the documentation on this. It has been fixed in 1.0.5.

**Comment by Dipak26** [comment]
How to remove placeholder number in ionic. I tried enablePlaceHolder to false but it not work
